sike.go

```diff
@@ -0,0 +1,174 @@
+/*
+Abstract:
+This file contains APIs for Supersingular Isogeny Key Encapsulation (SIKE) protocol which is recently submitted to NIST PQC candidates.
+The underlying core functions are based on SIDH keygeneration functions developed by Cloudflare.
+SIKE requires NIST's approved hash functions such as sha3 and cSHAKE256 to encapsulate the key based on NIST specifications. The custome 
+SHAKE256 is adopted from  https://gist.github.com/mimoo/7e815318e54d5c07c3330149ddf439c5
```

Comment:
This isn't going to work as-is because this requires the user to copy a file into an existing package before compiling. Can you either sha3 from crypto/sha3 and put a commit on top to incorporate this gist so it can be used?

Reply:
Unfortunately, I couldn't find any cSHAKE256 implementation inside crypto/sha3 and since SIKE requires custom SHAKE256, I had to use the gist. I think it's better to incorporate the gist in the next commit.